File: 1297-maximum-number-of-balloons/maximum-number-of-balloons.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        cnt = Counter(text)
        ans = []

        for k, v in cnt.items():
            if k == "b" or k == "a" or k == "n":
                ans.append(v)
            if k == "l" or k == "o":
                ans.append(v // 2)

        return min(ans) if len(ans) == 5 else 0

File: 1297-maximum-number-of-balloons/maximum-number-of-balloons.go
func maxNumberOfBalloons(text string) int {
	cnt := make(map[rune]int)

	for _, ch := range text {
		cnt[ch]++
	}

	cnt['l'] /= 2
	cnt['o'] /= 2

	ans := cnt['b']
	for _, ch := range []rune{'a', 'n', 'l', 'o'} {
		if cnt[ch] < ans {
			ans = cnt[ch]
		}
	}

	return ans
}